### ISSUE TITLE
support for disabling automagic unload/load to honor the one-name rule, and throw an error instead

### ISSUE THREAD

**Issue by boegel** [TACC/Lmod#48]
Currently, Lmod will automagically unload a loaded module with the same name as the module being loaded. 

Example:

```
$ module load EasyBuild/1.11.0 
$ module load EasyBuild/1.15.2

The following have been reloaded with a version change:
  1) EasyBuild/1.11.0 => EasyBuild/1.15.2
```

This makes sense, because loading two modules with the same name is stupid (as @rtmclay would say ;-)).

This does have one side-effect though: if these modules contain a `conflict EasyBuild` statement, it will not be honoured in some sense: no "conflict error" will be generated, Lmod will simply do the magic unload/load instead.

Would it be possible to support a configure option in Lmod such that the magic unload/load behaviour is changed to the stricter behaviour of throwing an error instead (regardless of whether there's a conflict statement in the module file(s))?

Maybe the error message should be something like this:

```
$ module load EasyBuild/1.11.0
$ module load EasyBuild/1.15.2

EasyBuild/1.15.2 can not be loaded, since a module with the same name is already loaded (EasyBuild/1.11.0).
Use "module swap EasyBuild/1.15.2" instead, or unload EasyBuild/1.11.0 first.
```

This would allow for syncing Lmod behaviour with our current setup, i.e. using Tcl/C environment modules and a conflict on module name in all of our module files.

**Comment by boegel** [comment]
A suggestion for configure option: `--one-name-rule-error=yes` (with the default being `no`)

**Comment by rtmclay** [comment]
this has been added.

**Comment by boegel** [comment]
Thanks Robert!

Maybe some context for people that stumble by here via Google (i.e., me, in the future ;-)):

the configure option that allows tweaking Lmod's behaviour as proposed is available since Lmod v5.8.5 as `--with_disableNameAutoSwap`, and however Lmod was configured it can be overridden via `$LMOD_DISABLE_SAME_NAME_AUTOSWAP ` at runtime
